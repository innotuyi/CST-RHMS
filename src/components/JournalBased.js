
import React, { useState } from "react";
import axios from 'axios';
import { useHistory } from "react-router-dom";
import {
  Row,
  Col,
  Container,
  Form,
  Button,
  Jumbotron,
} from "react-bootstrap";
import Header from "./Header";

const JournalBased = () => {
 const history = useHistory();
 const [title, setTitle] = useState("");
 const [author, setAuthor] = useState("");
 const [journel, setJournel] = useState("");
 const [volume, setVolume] = useState("");
 const [series, setSeries] = useState("");
 const [publsher, setPublisher] = useState("");
 const [pages, setPages] = useState("");
 const [issn, setIssn] = useState("");
 const [factor, setFactor] = useState("");
 const submitHandler = (e) => {
   e.preventDefault();
   const payload = {
    'title': title, 
    'author':author,
    'journal':journel,
    'Volume':volume,
    'series':series,
    'publisher': publsher, 
    'pages':pages,
    'issn':issn,
    'impact_factor':factor,
    'owner':1,
     'research':'PRJA1'}

axios.post('http://127.0.0.1:8000/journal/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});
history.push('/Journal');
   console.warn(title, author, journel, volume, series, publsher, pages, issn, factor);
 };

    return (
      <div
        style={{
          background: "linear-gradient(to bottom, white, 84%, #1089AF)",
        }}
      >
        <Header />
        <Container>
          <Row className="d-flex justify-content-center arighn-item-center">
            <Col className="d-flex justify-content-center">
              <Jumbotron
                style={{
                  background:
                    "linear-gradient(to bottom, white , 80%, #1089AF )",
                  borderRadius: "10px",
                  width: "90%",
                }}
              >
                <Form className="form" style={{}} onSubmit={submitHandler}>
                  <h2 className="text-center mb-5 text-secondary">
                    Journal Based Research Information
                  </h2>
                  <Form.Group>
                    <Form.Label>Title</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="title"
                      value={title}
                    onChange={(e) => setTitle(e.target.value)}
                      placeholder="Enter The Title Of Journel"
                    />
                  </Form.Group>
                  <Form.Group>
                    <Form.Label>Author</Form.Label>
                    <Form.Control
                      required
                      type="email"
                      id="author"
                     value={author}
                     onChange={(e) => setAuthor(e.target.value)}
                      placeholder="Enter The Author Email"
                    />
                  </Form.Group>
                  <Form.Group>
                    <Form.Label>Journal</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="journel"
                    value={journel}
                    onChange={(e) => setJournel(e.target.value)}
                      placeholder="Enter The Journel"
                    />
                  </Form.Group>

                  <Form.Group>
                    <Form.Label>Volume</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="volume"
                    value={volume}
                    onChange={(e) => setVolume(e.target.value)}
                      placeholder=""
                    />
                  </Form.Group>

                  <Form.Group>
                    <Form.Label>series</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="series"
                    value={series}
                    onChange={(e) => setSeries(e.target.value)}
                      placeholder=""
                    />
                  </Form.Group>

                  <Form.Group>
                    <Form.Label>Publisher</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="publsher"
                      value={publsher}
                    onChange={(e) => setPublisher(e.target.value)}
                      placeholder="Enter Publisher Name"
                    />
                  </Form.Group>

                  <Form.Group>
                    <Form.Label>Pages</Form.Label>
                    <Form.Control
                      required
                      type="number"
                      id="pages"
                    value={pages}
                    onChange={(e) => setPages(e.target.value)}
                      placeholder=""
                    />
                  </Form.Group>
                  <Form.Group>
                    <Form.Label>ISSN</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="issn"
                      value={issn}
                    onChange={(e) => setIssn(e.target.value)}
                      placeholder=""
                    />
                  </Form.Group>

                  <Form.Group>
                    <Form.Label>Impact factor</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="factor"
                    value={factor}
                    onChange={(e) => setFactor(e.target.value)}
                      placeholder=""
                    />
                  </Form.Group>      
                  <Col className="text-center">
                    <Button className="" type="submit">
                      Create
                    </Button>
                  </Col>
                </Form>
              </Jumbotron>
            </Col>
          </Row>
        </Container>
      </div>
    );
}

export default JournalBased;