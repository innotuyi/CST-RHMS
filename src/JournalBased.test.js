import React from "react";
import JournalBased from "./components/JournalBased";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<JournalBased />);
});

describe("Journal Based Tests", () => {
  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
