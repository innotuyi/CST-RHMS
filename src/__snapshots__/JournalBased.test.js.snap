// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal Based Tests matches Snapshot Collectly 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
    }
  }
>
  <Header />
  <Container
    fluid={false}
  >
    <Row
      className="d-flex justify-content-center arighn-item-center"
      noGutters={false}
    >
      <Col
        className="d-flex justify-content-center"
      >
        <Jumbotron
          fluid={false}
          style={
            Object {
              "background": "linear-gradient(to bottom, white , 80%, #1089AF )",
              "borderRadius": "10px",
              "width": "90%",
            }
          }
        >
          <Form
            className="form"
            inline={false}
            style={Object {}}
          >
            <h2
              className="text-center mb-5 text-secondary"
            >
              Journal Based Research Information
            </h2>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Conference name
              </FormLabel>
              <FormControl
                placeholder="Enter The Name Of The Conference"
                required={true}
                type="text"
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Conference Theme
              </FormLabel>
              <FormControl
                placeholder="Enter Conference Theme "
                required={true}
                type="text"
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Orgnizer's Name
              </FormLabel>
              <FormControl
                placeholder="Enter Orgnizer's Name"
                required={true}
                type="text"
              />
            </FormGroup>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId="dob"
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Select Conference Date
                  </FormLabel>
                  <FormControl
                    name="dos"
                    placeholder="Conference Date"
                    type="date"
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                     Place
                  </FormLabel>
                  <FormControl
                    name="dos"
                    placeholder="Enter Conference Location"
                    type="text"
                  />
                </FormGroup>
              </Col>
            </Row>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId=""
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Chapter Pages
                  </FormLabel>
                  <FormControl
                    name=""
                    placeholder="Enter The Number Of Pages"
                    type="Number"
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    ISBN
                  </FormLabel>
                  <FormControl
                    name=""
                    placeholder="Enter ISBN"
                    type="Number"
                  />
                </FormGroup>
              </Col>
            </Row>
            <Col
              className="text-center"
            >
              <Button
                active={false}
                className=""
                disabled={false}
                type="submit"
                variant="primary"
              >
                Create
              </Button>
            </Col>
          </Form>
        </Jumbotron>
      </Col>
    </Row>
  </Container>
</div>
`;
